src/search/FindInFilesUI.js

```diff
@@ -102,10 +107,45 @@ define(function (require, exports, module) {
 
                 StatusBar.hideBusyIndicator();
             })
+            .progress(function (results) {
+                // Update first page of panel (100 matches) for every file with matches,
+                // then only update the summary for every page (100 matches)
+                if (results.matches && results.matches.length) {
+                    progressCount += results.matches.length;
+                    if (!progressFirstUpdate) {
+                        // First update
+                        _resultsView.open();
+                        if (_findBar) {
+                            _findBar.close();
+                        }
+                        progressFirstUpdate = true;
+                    } else if (!progressFirstPage) {
+                        // First page
+                        if (progressCount >= 100) {
+                            progressCount -= 100;
+                            progressFirstPage = true;
+                        }
+                        _resultsView._updateResults();
+                    } else {
+                        // Subsequent full pages
+                        if (progressCount >= 100) {
+                            progressCount -= 100;
+                            _resultsView._showSummary();
```

Comment:
I still noticed that clicking on next is kind of hard since the buttons are constantly being updated. So we can do something fancier than rendering the summary and just update the text and the state of the buttons.